src/illumidesk/authenticators/authenticator.py

```diff
@@ -165,38 +165,39 @@ async def authenticate(self, handler: BaseHandler, data: Dict[str, str] = None)
         if validator.validate_launch_request(launch_url, handler.request.headers, args):
             # get the lms vendor to implement optional logic for said vendor
             lms_vendor = ''
-            if (
-                'tool_consumer_info_product_family_code' in args
-                and args['tool_consumer_info_product_family_code'] is not None
+            if 'tool_consumer_info_product_family_code' in args.keys() and args.get(
+                'tool_consumer_info_product_family_code'
             ):
                 lms_vendor = args['tool_consumer_info_product_family_code']
 
             # We use the course_id to setup the grader service notebook. Since this service
             # runs as a docker container we need to normalize the string so we can use it
             # as a container name.
-            if 'context_label' in args and args['context_label'] is not None:
-                course_id = args['context_label']
+            if 'context_label' in args.keys() and args.get('context_label'):
```

Comment:
I'm not sure why the other way was wrong, can you give me more context or explain what was the error you noticed?
I understand that the result is the same but I (IMHO) prefer the simplified form `if key in dict:`

Reply:
The other way wasn't wrong in the sense that the conditional to check if the key was in the dict was there or not, this issue was rather checking for `is not None`, since sometimes the value was set to an empty string instead of a None value. In those cases, the value still isn't applicable and shouldn't be set.

Regarding the use of `if 'foo' in args` instead of `if foo in args.keys()` it was set this way since the LTI13 authenticator already used that convention, but would agree that `if key in args` is simpler to understand, so I'll update the conditionals [with this more standard convention](https://docs.python.org/3/tutorial/datastructures.html#dictionaries).